src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -570,9 +570,20 @@ private boolean isParticipantInState(XWikiDocument regDoc, String emailAdr,
 
   @Override
   public boolean sendConfirmationMail(DocumentReference regDocRef, int participantObjNb) {
-    return XWikiObjectFetcher.on(modelAccess.getOrCreateDocument(regDocRef))
-        .filter(participantObjNb).stream().findFirst()
-        .map(this::sendConfirmationMail).orElse(false);
+    XWikiDocument regDoc = modelAccess.getOrCreateDocument(regDocRef);
+    boolean sendSuccess = XWikiObjectFetcher.on(regDoc).filter(participantObjNb).stream()
+        .findFirst().map(this::sendConfirmationMail).orElse(false);
+    if(sendSuccess)  {
```

Comment:
1. do not change semantic of sendConfirmationMail. Not all sending of a confirmation Mail may change the ParticipantStatus to confirmed. Only if a confirmation mail is send trough the backend the status may change to confirmed if it is unconfirmed
2. ensure that a status of payed, cancelled and duplicate is not changed. Only the "unconfirmed" may change to confirmed.

Reply:
I agree, I wouldn't expect `sendConfirmationMail` to alter the registration doc as a client. I'd rather see an additional explicit call in `sendParticipantConfirmationMail.vm`.